src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -1444,4 +1454,42 @@ protected function get_edit_media_item_args() {
 		);
 	}
 
+	/**
+	 * Issue an incorrect-usage error when a featured image has been assigned to an attachment with
+	 * a MIME type that doesn't support post thumbnails.
+	 *
+	 * @see https://core.trac.wordpress.org/ticket/41692
+	 *
+	 * @param int    $attachment_id Attachment ID.
+	 * @param string $method        Method name.
+	 */
+	private function maybe_did_featured_media_wrong( $attachment_id, $method ) {
+		$post = get_post( $attachment_id );
+
+		if ( ! $post instanceof WP_Post || 'attachment' !== $post->post_type ) {
+			return;
+		}
+
+		$thumbnail_support = current_theme_supports( 'post-thumbnails', $post->post_type ) && post_type_supports( $post->post_type, 'thumbnail' );
+
+		if ( false === $thumbnail_support && '' !== $post->post_mime_type ) {
+			if ( wp_attachment_is( 'audio', $post ) ) {
+				$thumbnail_support = post_type_supports( 'attachment:audio', 'thumbnail' ) || current_theme_supports( 'post-thumbnails', 'attachment:audio' );
+			} elseif ( wp_attachment_is( 'video', $post ) ) {
+				$thumbnail_support = post_type_supports( 'attachment:video', 'thumbnail' ) || current_theme_supports( 'post-thumbnails', 'attachment:video' );
+			}
+		}
+
+		if ( false === $thumbnail_support ) {
+			_doing_it_wrong(
+				$method,
+				sprintf(
+					/* translators: %s: attachment mime type */
+					__( 'This site does not support post thumbnails on attachments with MIME type %s.' ),
+					$post->post_mime_type
+				),
+				'6.2.0'
+			);
```

Comment:
return WP_Error.

Reply:
Can you explain more? Why return a `WP_Error` here?